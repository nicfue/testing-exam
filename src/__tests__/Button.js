import React from 'react';
import { mount, shallow, render } from 'enzyme';
import Button from '../components/Button';

describe('button test', () => {
  const btn = shallow(<Button onClick={() => {}} />);
  const btnClick = jest.fn();
  const button = shallow(<Button onClick={btnClick}></Button>)
  const dangerButton = shallow(<Button danger onClick={btnClick}></Button>)

 it('check if button exists', () => {
   expect(btn.exists()).toBeTruthy();
 });

 it('check if button works onClick', () => {
     button.simulate('click');
     expect(btnClick).toBeDefined();
 })
 it('button snapshot should match default button', () => {
   expect(button).toMatchSnapshot();
 })
 it('button snapshot should match default button', () => {
   expect(dangerButton).toMatchSnapshot();
 })
});
