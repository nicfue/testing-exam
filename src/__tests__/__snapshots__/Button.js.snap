// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test button snapshot should match default button 1`] = `
<button
  className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full "
  data-test="button"
  onClick={[Function]}
/>
`;

exports[`button test button snapshot should match default button 2`] = `
<button
  className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full "
  data-test="button"
  onClick={[Function]}
/>
`;
